### PR TITLE
[Google Fonts] Fix loading fonts from assets and offline behavior

### LINKED ISSUE #28: Provide option to check to use fonts in assets and/or pubspec first (rather than fetching from URL)
For example, when calling GoogleFonts.foo(), the library checks to see if the font has been bundled via the assets folder and/or pubspec (either in AssetManifest.json or FontManifest.json), before fetching it from the URL

### LINKED ISSUE #31: App Crashed if no Internet
When no Internet connection is available, App is crashed with following error on iOS:

```
[VERBOSE-2:ui_dart_state.cc(157)] Unhandled Exception: SocketException: Failed host lookup: 'fonts.gstatic.com' (OS Error: nodename nor servname provided, or not known, errno = 8)
#0      IOClient.send (package:http/src/io_client.dart:33:23)           
<asynchronous suspension>                                               
#1      BaseClient._sendUnstreamed (package:http/src/base_client.dart:169:38)
#2      BaseClient.get (package:http/src/base_client.dart:32:7)         
#3      _httpFetchFont (package:google_fonts/src/google_fonts_base.dart:140:37)
#4      loadFontIfNecessary (package:google_fonts/src/google_fonts_base.dart:99:16)
<asynchronous suspension>                                               
#5      googleFontsTextStyle (package:google_fonts/src/google_fonts_base.dart:64:3)
#6      GoogleFonts.poppins (package:google_fonts/google_fonts.dart:32044:12)
#7      GoogleFonts.poppinsTextTheme (package:google_fonts/google_fonts.dart:32057:29)
#8      main (package:codescanner/main.dart:18:32)                      
#9      _runMainZoned.<anonymous closure>.<anonymous closure> (dart:ui/hooks.dart:239:25)
#10     _rootRun (dart:async/zone.dart:1126:13)                         
#11     _CustomZone.run (dart:async/zone.dart:1023:19)                  
#12     _runZoned (dart:async/zone.dart:1518:10)                        
#13     runZoned (dart:async/zone.dart:1502:12)                         
#14     _runMainZoned.<anonymous closure> (dart:ui/hooks.dart:231:5)    
#15     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:307:19)
#16     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174:12)
```

If no Internet is available, it should default to system font (like it does on Android).

### LINKED ISSUE #40: Add build/testfile.dill.track.dill to gitignore

### PR DESCRIPTION
**Pull request by guidezpl**
Fixes asset font loading and gracefully handles exception cases (i.e. offline) by printing errors to console. This PR also updates the bundling instructions and removes files which shouldn't be checked-in.

## Related issues
Closes #40 
Closes #31
Closes #28
#49
#54
#38

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,10 +117,14 @@ filenames as follows:
 }
 ```
 
-Finally, make sure you have listed the `google_fonts` folder in your `pubspec.yaml`.
+2. Move those fonts to a top-level app directory (e.g. `google_fonts`).
+
+![](https://raw.githubusercontent.com/material-foundation/google-fonts-flutter/master/readme_images/google_fonts_folder.png)
+
+3. Ensure sure you have listed the folder (e.g. `google_fonts/`) in your `pubspec.yaml` under `assets`.
```
Comment (clocksmith) [inline README.md]: Ensure that you have

File: lib/src/google_fonts_base.dart [hunk 2]
```diff
@@ -120,31 +119,50 @@ Future<void> loadFontIfNecessary(GoogleFontsDescriptor descriptor) async {
...
-    return;
-  }
+  try {
+    // Check if this font can be loaded by the pre-bundled assets.
+    final assetManifestJson = await _loadAssetManifestJson();
+    final assetPath = _findFamilyWithVariantAssetPath(
+      descriptor.familyWithVariant,
+      assetManifestJson,
+    );
+    if (assetPath != '') {
```
Comment (clocksmith) [inline lib/src/google_fonts_base.dart]: should we use null here instead of empty string as sentinel value?

File: lib/src/google_fonts_base.dart [hunk 3]
```diff
@@ -253,11 +276,15 @@ Future<Map<String, dynamic>> _loadAssetManifestJson() async {
   }
 }
 
-bool _isFamilyWithVariantInManifest(
+/// Looks for a matching [familyWithVariant] font, provided the asset manifest.
+/// Returns the path of the font asset if found, otherwise an empty string.
```
Comment (clocksmith) [inline lib/src/google_fonts_base.dart]: Maybe this should return null?

Comment (guidezpl) [reply lib/src/google_fonts_base.dart]: Done

File: lib/src/google_fonts_base.dart (hunk 2 above)
Comment (guidezpl) [reply lib/src/google_fonts_base.dart]: I like that better!

**Comment by Zazo032** [comment]
@guidezpl you'll need to solve merge conflicts so this PR can be merged

**Pull request merged**